tests/arithmetic/ast_arithmetic.re

```diff
@@ -0,0 +1,9 @@
+/* Auto-generated by codegen_type */
+type assignment_statement = (variable, string, expression, string)
+and comment = string
+and expression = Intermediate_type1(variable) | Intermediate_type2(number) | Intermediate_type3((expression, string, expression)) | Intermediate_type4((expression, string, expression)) | Intermediate_type5((expression, string, expression)) | Intermediate_type6((expression, string, expression)) | Intermediate_type7((expression, string, expression))
```

Comment:
could be nice at some point to choose a name of a constructor deriving from the type, for example here
and expression = Variable1(variable) | Number(number) | Misc1(expression, string, expression) | Misc2(...)

Reply:
I will do that here

Reply:
This is not trivial, we can use the AST normalized node in conjunction with the identify string to get a nice constructor name going. I will make that a separate PR with regexes